go/host/host.go

```diff
@@ -1074,3 +1093,20 @@ func (h *host) validateConfig() {
 		h.logger.Crit("the host must specify a public P2P address")
 	}
 }
+
+// this function should be fired off in a new goroutine whenever the status of the enclave needs to be verified
+// (e.g. if we've seen unexpected errors from the enclave client)
+func (h *host) checkEnclaveStatus() {
+	// only allow one status request at a time, if lock unavailable then abandon attempt
+	if h.statusReqLock.TryLock() {
```

Comment:
<img width="965" alt="image" src="https://github.com/obscuronet/go-obscuro/assets/4471462/cf5129fd-51e9-4dca-a45b-245e1521b292">

I don't see anything wrong with it per-say but given the documentations and purely from a conservative POV I would probably suggest using an atomic int instead.

```go
if !atomic.CompareAndSwapUint32(&lock, 0, 1) {
        return  // it's locked
    }                                               
defer atomic.StoreUint32(&locker, 0)          
doTheThing()
```